interfaces/hotplug/spec.go

```diff
@@ -0,0 +1,70 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package hotplug
+
+import (
+	"fmt"
+
+	"github.com/snapcore/snapd/interfaces/utils"
+)
+
+type SlotSpec struct {
+	Name  string
+	Label string
+	Attrs map[string]interface{}
+}
+
+type Specification struct {
+	slots map[string]SlotSpec
+}
+
+func NewSpecification() *Specification {
+	return &Specification{
+		slots: make(map[string]SlotSpec),
+	}
+}
+
+func (h *Specification) AddSlot(name, label string, attrs map[string]interface{}) error {
```

Comment:
Can this take a SlotInfo instead (and ensure some things are not wired, i.e. no snap.Info). This is more future proof than a function with three arguments.